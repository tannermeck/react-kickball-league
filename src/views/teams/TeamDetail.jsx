import { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { getTeamById } from '../../services/teams';
import './teamdetail.css';

function TeamDetail({ match }){
    const { id } = match.params
    const [team, setTeam] = useState({});
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        async function loadTeams(){
            const teamDetail = await getTeamById(id)
            setTeam(teamDetail)
            setLoading(false)
        }
        loadTeams()
    }, [id])
    if (loading) return <h1>Loading Team Details...</h1>

    return (
        <>
            <h1 className='detailHeader'>{team.name}</h1>
            <h3 className='detailHeader'>Location: {team.city}, {team.state}</h3>
            <NavLink to={`/teams/edit/${team.id}`} >{`Edit ${team.name}`}</NavLink>
            <h1 className='detailHeader'>Players:</h1>
            <div className='playerDetail'>
                {team.players.map(player => (                   
                    <h3 key={player.id}>{player.position}: 
                    <NavLink key={player.id} to={`/players/${player.id}`}>
                        {player.name}
                    </NavLink></h3> 
                ))}
                </div>
        </>
    )
}
export default TeamDetail;