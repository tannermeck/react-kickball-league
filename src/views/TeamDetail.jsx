import { useState, useEffect } from 'react';
import { NavLink, useParams } from 'react-router-dom';
import { getTeamById } from '../services/teams';

function TeamDetail(){
    const { id } = useParams();
    const [team, setTeam] = useState({});
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        async function loadTeams(){
            const teamDetail = await getTeamById(id)
            setTeam(teamDetail)
            setLoading(false)
        }
        loadTeams()
    }, [id])
    console.log(team)
    if (loading) return <h1>Loading Team Details...</h1>

    return (
        <>
            <h1>{team.name}</h1>
            <h3>Location: {team.city}, {team.state}</h3>
            <h1>Players:</h1>
                {team.players.map(player => (
                    <NavLink key={player.id} to={`/players/${player.id}`}>
                        <h3 key={player.id}>{player.name}- {player.position}</h3>
                    </NavLink>
                ))}
        </>
    )
}
export default TeamDetail;