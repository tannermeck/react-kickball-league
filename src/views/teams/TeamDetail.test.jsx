import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import TeamDetail from './TeamDetail';

it('should render out a team detail component', async () => { 
    const { container } = render(
                            <MemoryRouter>
                                <TeamDetail match={{ params: { id: '1'} }}/>
                            </MemoryRouter>)            
    const label = await screen.findByText('Players:', {exact: false})
    
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    
})