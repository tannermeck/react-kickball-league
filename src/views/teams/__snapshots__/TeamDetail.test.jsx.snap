// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render out a team detail component 1`] = `
<div>
  <h1
    class="detailHeader"
  >
    My Team
  </h1>
  <h3
    class="detailHeader"
  >
    Location: 
    Portland
    , 
    Or
  </h3>
  <a
    href="/teams/edit/1"
  >
    Edit My Team
  </a>
  <h1
    class="detailHeader"
  >
    Players:
  </h1>
  <div
    class="playerDetail"
  />
</div>
`;
